### ISSUE TITLE
LLVM crash from EOSIO

### ISSUE THREAD

**Issue by fcecin** [llvm/llvm-project#60564]
LLVM is used in the EOSIO blockchain smart contract compilation suite for its WASM VM.

I got the following "PLEASE submit this bug report" message from the EOSIO contract compiler while trying to compile a massive contract that is basically a doctored version of the entire Lua 5.4.4 language distribution.

I have no idea if this report is actually useful to the LLVM project -- I have no real need for this to be fixed -- but anyway here is the report and the associated bug report files (cpp, sh) attached in a ZIP file here:

[lmc-64e907.zip](https://github.com/llvm/llvm-project/files/10665360/lmc-64e907.zip)

This is the tool output I get which asks me to submit this report:

```
eosio-cpp lmc.cpp -o lmc.wasm -I .
Warning, empty ricardian clause file
Warning, empty ricardian clause file
Warning, action <newstate> does not have a ricardian contract
Warning, action <delstate> does not have a ricardian contract
Warning, action <run> does not have a ricardian contract
clang-9: error: unable to execute command: Killed
clang-9: error: clang frontend command failed due to signal (use -v to see invocation)
clang version 9.0.1 (https://github.com/eosio/clang.git 6a2b43d5cb01f666af0dbfb31a3cbf2e17967909) (https://github.com/eosio/llvm d700c78db1918c7a2783c37a97d42d6fcd223e4d)
Target: wasm32
Thread model: posix
InstalledDir: /home/fcecin/lmc
clang-9: note: diagnostic msg: PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace, preprocessed source, and associated run script.
clang-9: note: diagnostic msg: 
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
clang-9: note: diagnostic msg: /tmp/lmc-64e907.cpp
clang-9: note: diagnostic msg: /tmp/lmc-64e907.sh
clang-9: note: diagnostic msg: 

********************
```
